packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/IconLabelIncrementalLayoutEngine.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams.layout.incremental;
+
+import java.util.EnumSet;
+import java.util.Optional;
+
+import org.eclipse.elk.core.math.ElkPadding;
+import org.eclipse.elk.core.math.KVector;
+import org.eclipse.elk.core.options.CoreOptions;
+import org.eclipse.elk.core.options.SizeConstraint;
+import org.eclipse.elk.graph.properties.IPropertyHolder;
+import org.eclipse.sirius.components.diagrams.NodeType;
+import org.eclipse.sirius.components.diagrams.Position;
+import org.eclipse.sirius.components.diagrams.Size;
+import org.eclipse.sirius.components.diagrams.Size.Builder;
+import org.eclipse.sirius.components.diagrams.events.IDiagramEvent;
+import org.eclipse.sirius.components.diagrams.layout.ISiriusWebLayoutConfigurator;
+import org.eclipse.sirius.components.diagrams.layout.incremental.data.LabelLayoutData;
+import org.eclipse.sirius.components.diagrams.layout.incremental.data.NodeLayoutData;
+import org.springframework.stereotype.Service;
+
+/**
+ * The incremental layout engine to layout icon label nodes.
+ *
+ * @author gcoutable
+ */
+@Service
+public class IconLabelIncrementalLayoutEngine implements INodeIncrementalLayoutEngine {
+
+    public IconLabelIncrementalLayoutEngine(INodeIncrementalLayoutEngineRegistry registry) {
+        registry.add(this);
+    }
```

Comment:
That registry is becoming more and more fishy... We know have a constructor for a stateless service which creates a side effect. I don't see why you need those registries.

Reply:
The registry has been removed